services/provider/server/server.go

```diff
@@ -206,3 +212,131 @@ func Start(port int, providerServer *OCSProviderServer, opts []grpc.ServerOption
 		klog.Fatalf("failed to start gRPC server: %v", err)
 	}
 }
+
+func (s *OCSProviderServer) getJSONConnectionDetails(consumerResource *ocsv1alpha1.StorageConsumer) ([]externalSC.ExternalResource, error) {
+	var j []externalSC.ExternalResource
+
+	// Configmap with mon endpoints
+	configmap, err := s.consumerAPI.GetConfigMap("rook-ceph-mon-endpoints", s.namespace)
+	if err != nil {
+		return []externalSC.ExternalResource{}, fmt.Errorf("failed to get `rook-ceph-mon-endpoints` configMap. Error %v", err)
+	}
+
+	cmData := strings.Split(configmap.Data["data"], ",")
+	data := map[string]string{
+		"data":     cmData[0],
+		"maxMonId": "0",
+		"mapping":  "{}",
+	}
+	name := "rook-ceph-mon-endpoints"
+	kind := "ConfigMap"
+
+	j = append(j, externalSC.ExternalResource{Name: name, Kind: kind, Data: data})
+
+	// Secret storing cluster mon.admin key, fsid and name
+	scMon, err := s.consumerAPI.GetSecret("rook-ceph-mon", s.namespace)
+	if err != nil {
+		return []externalSC.ExternalResource{}, fmt.Errorf("failed to get `rook-ceph-mon` secret. Error %v", err)
+	}
+	data = map[string]string{
+		"fsid":         string(scMon.Data["fsid"]),
+		"mon-secret":   "mon-secret",
+		"admin-secret": "admin-secret",
+	}
+	name = "rook-ceph-mon"
+	kind = "Secret"
+
+	j = append(j, externalSC.ExternalResource{Name: name, Kind: kind, Data: data})
+
+	// Secret storing CSI RBD node userID and key
+	scCsiRbdNode, err := s.consumerAPI.GetSecret("rook-csi-rbd-node", s.namespace)
```

Comment:
This secret does not exist in the provider cluster. What exists is a CephClient CRD that points to a Secret with the credentials.

Reply:
Don't you get this from `storageConsumer.Status.CephObjects` somewhere?

Reply:
The secrets will be created once Ceph client cr is created.

Reply:
@leseb yes, we can get the details from storageConsumer.Status.CephObjects. Also, does this secret value needs to be placed in both `csi rbd node` and `csi rbd provisioner` ?

Reply:
There are 4 secrets for ceph-csi:

* rook-csi-cephfs-node
* rook-csi-cephfs-provisioner
* rook-csi-rbd-node
* rook-csi-rbd-provisioner